cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -26,6 +27,15 @@ namespace raft {
 namespace matrix {
 namespace detail {
 
+namespace stdex    = std::experimental;
+using extents_type = stdex::extents<stdex::dynamic_extent, stdex::dynamic_extent>;
+template <typename ValueType>
+using padded_layout_row_major =
```

Comment:
Perhaps, this, parameterized by the padding size, should go to `raft/core/mdarray.hpp`?..